src/secure-postgresql-db/index.ts

```diff
@@ -0,0 +1,551 @@
+      allocatedStorage: this._props.allocatedStorage,
+      maxAllocatedStorage: this._props.maxAllocatedStorage,
+      cloudwatchLogsExports: ['postgresql', 'upgrade'],
+      cloudwatchLogsRetention: this._props.cloudWatchLogRetention,
+      credentials: Credentials.fromSecret(this.credentialSecret),
+      parameterGroup: createRDSLogEnabledParamGroup(this, {
+        postgresEngineVersion: this._props.postgresEngineVersion!,
+      }),
+      removalPolicy: RemovalPolicy.RETAIN,
+      storageEncrypted: true,
+      storageEncryptionKey: this.encryptionKey,
+      backupRetention: Duration.days(7),
+      ...this._props.overrideInstanceProps,
+    });
+
+    if (this._props.snapshotArn) {
+      const cfnDbInstance = this.instance.node.defaultChild as CfnDBInstance;
+      cfnDbInstance.addPropertyOverride('DbSnapshotIdentifier', this._props.snapshotArn);
+    }
+  }
+
+  private _createEncryptionKey() {
+    this.encryptionKey = new Key(this, 'DbKmsKey', {
+      description: `CDK generated database encryption key. STACK - ${cdk.Stack.of(this)}`,
+      enableKeyRotation: true,
+    });
+  }
+
+  private _createSecret() {
+    this.credentialSecret = new Secret(this, 'DbCredentialSecret', {
+      generateSecretString: {
+        secretStringTemplate: JSON.stringify({ username: 'postgres' }),
+        excludePunctuation: true,
+        includeSpace: false,
+        generateStringKey: 'password',
+      },
+    });
+  }
+
+  private _createSecurityGroup() {
+    this.securityGroup = new SecurityGroup(this, 'DbSecurityGroup', { vpc: this._vpc });
+
+    this._allowedSgs.forEach((sg) => {
+      this._addSgIngress(sg);
+    });
+  }
+
+  private _createSSMParams() {
+    super.setParameter(
+      'DbCredentialsSecretArn',
+      `/cdk/${cdk.Stack.of(this)}/db/credentials-secret-arn`,
+      this.credentialSecret.secretArn,
+    );
+    super.setParameter(
+      'DbPort',
+      `/cdk/${cdk.Stack.of(this)}/db/port`,
+      this._props.port?.toString()!,
+    );
+    super.setParameter(
+      'DbHost',
+      `/cdk/${cdk.Stack.of(this)}/db/host`,
+      this.instance.dbInstanceEndpointAddress);
+    super.setParameter(
+      'DbId',
+      `/cdk/${cdk.Stack.of(this)}/db/id`,
+      this.instance.instanceIdentifier,
+    );
+    super.setParameter(
+      'DBSecurityGroup',
+      `/cdk/${cdk.Stack.of(this)}/db/sg-id`,
+      this.securityGroup.securityGroupId,
+    );
+  }
+
+  private _suppressNagRules() {
+    Utils.suppressNagRule(this.instance, 'AwsSolutions-RDS11', 'Ignore non default port(5432) usage requirement.');
+    Utils.suppressNagRule(this.credentialSecret, 'AwsSolutions-SMG4', 'Ignore password automatic rotation.');
+  }
+
+  public assemble(): SecurePostgresqlDb {
+    this._vpc = Vpc.fromLookup(this, 'DbVpc', { vpcId: this._props.vpcId });
+
+    this._createEncryptionKey();
+    this._createSecret();
+    this._createSecurityGroup();
+    this._createDbInstance();
+    this._createBackupPlan();
```

Comment:
We should not be automatically enforcing this backup plan. The idea is to utilize the backup plan that is globally available via organization. 

Let the "_createBackupPlan()" method be there, do not remove that.

Reply:
Added `enableBackupPlan` prop / method which defaults to false